arch/src/main/java/knaufdan/android/arch/mvvm/implementation/dialog/BaseDialogFragment.kt

```diff
@@ -0,0 +1,107 @@
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ViewDataBinding
+import androidx.fragment.app.DialogFragment
+import androidx.lifecycle.ViewModelProvider
+import dagger.android.support.AndroidSupportInjection
+import javax.inject.Inject
+import knaufdan.android.arch.dagger.vm.ViewModelFactory
+import knaufdan.android.arch.mvvm.IBaseFragment
+import knaufdan.android.arch.mvvm.implementation.BaseViewModel
+import knaufdan.android.arch.mvvm.implementation.Config
+
+abstract class BaseDialogFragment<ViewModel : BaseViewModel> : DialogFragment(),
+    IBaseFragment<ViewModel> {
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+
+    private lateinit var viewModel: ViewModel
+
+    override fun getDataSource(): ViewModel = viewModel
+
+    private val config: Config.DialogConfig by lazy {
+        Config.DialogConfig(
+            layoutRes = getLayoutRes(),
+            viewModelKey = getBindingKey(),
+            titleRes = getTitleRes(),
+            dialogStyle = getDialogStyle()
+        )
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        AndroidSupportInjection.inject(this)
+
+        viewModel = ViewModelProvider(this, viewModelFactory).get(getDataSourceClass())
+        viewModel.fragmentTag = getFragmentTag()
+        lifecycle.addObserver(viewModel)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? =
+        config.run {
+            if (titleRes != -1) {
+                activity?.setTitle(titleRes)
+            }
+
+            // do only initiate view model on first start
+            if (savedInstanceState == null) {
+                viewModel.handleBundle(arguments)
+            }
+
+            val binding: ViewDataBinding =
+                DataBindingUtil.inflate(
+                    inflater,
+                    layoutRes,
+                    container,
+                    false
+                )
+
+            binding.run {
+                setVariable(viewModelKey, viewModel)
+                executePendingBindings()
+                binding.root
+            }
+        }
+
+    override fun onStart() {
+        super.onStart()
+
+        dialog?.apply {
+            when (config.dialogStyle) {
+                DialogStyle.FULL_SCREEN -> applyFullScreen()
+                DialogStyle.FULL_WIDTH -> applyFullWidth()
+            }
+        }
+    }
+
+    private fun Dialog.applyFullScreen() {
+        val width = ViewGroup.LayoutParams.MATCH_PARENT
+        val height = ViewGroup.LayoutParams.MATCH_PARENT
+        window?.setLayout(width, height)
+    }
+
+    private fun Dialog.applyFullWidth() {
+        val width = ViewGroup.LayoutParams.MATCH_PARENT
+        val height = ViewGroup.LayoutParams.WRAP_CONTENT
+        window?.setLayout(width, height)
+    }
```

Comment:
Make a function taking `LayoutParams` for height and width and applying them to `window`